# Intro

- show code snippets
- autocomplete
- text/code cells
- share features
- exporting

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
!pip install pandas

import pandas as pd

path ="/content/drive/MyDrive/sample_sales_data.csv"
df = pd.read_csv(path)

#Dataset is now stored in a Pandas Dataframe

df.describe

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


<bound method NDFrame.describe of       Unnamed: 0                        transaction_id            timestamp  \
0              0  a1c82654-c52c-45b3-8ce8-4c2a1efe63ed  2022-03-02 09:51:38   
1              1  931ad550-09e8-4da6-beaa-8c9d17be9c60  2022-03-06 10:33:59   
2              2  ae133534-6f61-4cd6-b6b8-d1c1d8d90aea  2022-03-04 17:20:21   
3              3  157cebd9-aaf0-475d-8a11-7c8e0f5b76e4  2022-03-02 17:23:58   
4              4  a81a6cd3-5e0c-44a2-826c-aea43e46c514  2022-03-05 14:32:43   
...          ...                                   ...                  ...   
7824        7824  6c19b9fc-f86d-4526-9dfe-d8027a4d13ee  2022-03-03 18:22:09   
7825        7825  1c69824b-e399-4b79-a5e7-04a3a7db0681  2022-03-04 19:14:46   
7826        7826  79aee7d6-1405-4345-9a15-92541e9e1e74  2022-03-03 14:00:09   
7827        7827  e5cc4f88-e5b7-4ad5-bc1b-12a828a14f55  2022-03-04 15:11:38   
7828        7828  afd70b4f-ee21-402d-8d8f-0d9e13c2bea6  2022-03-06 13:50:36   

                 

In [ ]:
df[['unit_price', 'quantity', 'total']].describe()

unit_price     quantity        total
count  7829.000000  7829.000000  7829.000000
mean      7.819480     2.501597    19.709905
std       5.388088     1.122722    17.446680
min       0.190000     1.000000     0.190000
25%       3.990000     1.000000     6.570000
50%       7.190000     3.000000    14.970000
75%      11.190000     4.000000    28.470000
max      23.990000     4.000000    95.960000

At the 25th percentile, we find that the most frequent values for unit_price, quantity, and total are 3.99 or less, 1.00 or less, 6.57 or less respectively.
At the 50th percentile, we learn that the most frequent figures for unit_price, quantity, and total are 7.19 or less, 3.00 or less, and 14.97 or less respectively.
At the 75th percentile, we may appreciate that the most common values for unit_price, quantity, and total are 11.19 or less, 4.00 or less, and 28.47 or less.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7829 entries, 0 to 7828
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      7829 non-null   int64  
 1   transaction_id  7829 non-null   object 
 2   timestamp       7829 non-null   object 
 3   product_id      7829 non-null   object 
 4   category        7829 non-null   object 
 5   customer_type   7829 non-null   object 
 6   unit_price      7829 non-null   float64
 7   quantity        7829 non-null   int64  
 8   total           7829 non-null   float64
 9   payment_type    7829 non-null   object 
dtypes: float64(2), int64(2), object(6)
memory usage: 611.8+ KB


This is a concise summary statistics of the sample data, in which we find out the nature of the sample data. There are 7,829 observations, 10 attributes, and three data-types (float numbers, integers, and objects).

In [ ]:
df.head()

Unnamed: 0                        transaction_id            timestamp  \
0           0  a1c82654-c52c-45b3-8ce8-4c2a1efe63ed  2022-03-02 09:51:38   
1           1  931ad550-09e8-4da6-beaa-8c9d17be9c60  2022-03-06 10:33:59   
2           2  ae133534-6f61-4cd6-b6b8-d1c1d8d90aea  2022-03-04 17:20:21   
3           3  157cebd9-aaf0-475d-8a11-7c8e0f5b76e4  2022-03-02 17:23:58   
4           4  a81a6cd3-5e0c-44a2-826c-aea43e46c514  2022-03-05 14:32:43   

                             product_id category customer_type  unit_price  \
0  3bc6c1ea-0198-46de-9ffd-514ae3338713    fruit          gold        3.99   
1  ad81b46c-bf38-41cf-9b54-5fe7f5eba93e    fruit      standard        3.99   
2  7c55cbd4-f306-4c04-a030-628cbe7867c1    fruit       premium        0.19   
3  80da8348-1707-403f-8be7-9e6deeccc883    fruit          gold        0.19   
4  7f5e86e6-f06f-45f6-bf44-27b095c9ad1d    fruit         basic        4.49   

   quantity  total payment_type  
0         2   7.98     e-wallet  
1         1   3.99     e-wallet  
2         2   0.38     e-wallet  
3         4   0.76     e-wallet  
4         2   8.98   debit card

This is a high-level overview of the sample data, and we learn that there are 10 features in total and that total is the response features. Since we are working with qualitative data mostly, I suggest clustering the observations on a diagram by timestamp and total spent. Subsequently, I recommend pursuing the same strategy for customer type and total spent at the grocery store.

In [ ]:
df[['customer_type']].value_counts()

customer_type
non-member       1601
standard         1595
premium          1590
basic            1526
gold             1517
dtype: int64

Running the method count values for the feature (attribute) customer_type, we learn how many observations fall into the categories non-member, standard, premium, basic, and gold. The largest type of consumers at Gala Groceries are non-members, followed by standard, and premium.

In [ ]:
df[['customer_type']].value_counts(normalize=True)

customer_type
non-member       0.204496
standard         0.203730
premium          0.203091
basic            0.194916
gold             0.193767
dtype: float64

Adding the normalization boolean to the value count method produces a statistical description of the attribute customer_type.

In [ ]:
df[['category']].value_counts()

category             
fruit                    998
vegetables               846
packaged foods           507
baked goods              443
canned foods             431
refrigerated items       425
meat                     382
kitchen                  382
dairy                    375
beverages                301
cheese                   293
cleaning products        292
baking                   264
frozen                   263
snacks                   263
seafood                  253
medicine                 243
baby products            224
condiments and sauces    181
personal care            177
pets                     161
spices and herbs         125
dtype: int64

Repeting the same study for the category attribute. Fruit and vegetables are the two most frequently purchased items at Gala Groceries.

In [ ]:
df[['category']].value_counts(normalize=True)

category             
fruit                    0.127475
vegetables               0.108060
packaged foods           0.064759
baked goods              0.056584
canned foods             0.055052
refrigerated items       0.054285
meat                     0.048793
kitchen                  0.048793
dairy                    0.047899
beverages                0.038447
cheese                   0.037425
cleaning products        0.037297
baking                   0.033721
frozen                   0.033593
snacks                   0.033593
seafood                  0.032316
medicine                 0.031038
baby products            0.028612
condiments and sauces    0.023119
personal care            0.022608
pets                     0.020565
spices and herbs         0.015966
dtype: float64

Respresenting the frequency of purchases by category using percentages.

In [ ]:
df.groupby('category')[['unit_price']].mean()

unit_price
category                         
baby products           12.285536
baked goods              7.619797
baking                   5.342273
beverages               12.624551
canned foods             6.269350
cheese                   8.651775
cleaning products       12.802671
condiments and sauces    6.886685
dairy                    7.980133
frozen                  10.071369
fruit                    2.503627
kitchen                 15.100995
meat                    15.012513
medicine                17.148848
packaged foods           4.924714
personal care            9.198475
pets                    10.578820
refrigerated items       6.922235
seafood                 16.063518
snacks                   2.378593
spices and herbs         3.080400
vegetables               2.368369

Presenting the average unit cost per category at the store.

In [ ]:
df.groupby('category')[['unit_price']].median()

unit_price
category                         
baby products               12.99
baked goods                  7.99
baking                       5.99
beverages                   13.19
canned foods                 7.49
cheese                       6.49
cleaning products           14.19
condiments and sauces        7.99
dairy                        8.19
frozen                      11.19
fruit                        3.99
kitchen                     16.19
meat                        14.99
medicine                    15.49
packaged foods               4.99
personal care                8.49
pets                         9.49
refrigerated items           6.49
seafood                     18.19
snacks                       1.99
spices and herbs             3.99
vegetables                   1.99

Presenting the median cost per category at the store after revising the nature of the sample data.

In [ ]:
df.groupby('category')[['quantity']].mean().sort_values(by='quantity')

quantity
category                       
baby products          2.424107
spices and herbs       2.432000
fruit                  2.443888
pets                   2.453416
baking                 2.454545
meat                   2.458115
baked goods            2.458239
medicine               2.477366
snacks                 2.479087
cleaning products      2.482877
condiments and sauces  2.486188
kitchen                2.500000
refrigerated items     2.505882
frozen                 2.509506
canned foods           2.510441
vegetables             2.510638
packaged foods         2.524655
dairy                  2.533333
personal care          2.570621
beverages              2.578073
seafood                2.644269
cheese                 2.672355

Sorting the quantities of items per category at the store from the lowest to the highest using the mean measurement.

In [ ]:
df.groupby('category')[['quantity']].median().sort_values(by='quantity')

quantity
category                       
baby products               2.0
pets                        2.0
medicine                    2.0
meat                        2.0
condiments and sauces       2.0
cleaning products           2.0
fruit                       2.0
canned foods                2.0
baking                      2.0
baked goods                 2.0
kitchen                     2.5
dairy                       3.0
frozen                      3.0
spices and herbs            3.0
beverages                   3.0
packaged foods              3.0
personal care               3.0
refrigerated items          3.0
seafood                     3.0
snacks                      3.0
cheese                      3.0
vegetables                  3.0

Sorting the volume of items per category from the lowest to the highest using the median measurement.

In [ ]:
df.groupby('category')[['quantity']].agg(['mean', 'median', 'min', 'max'])

quantity               
                           mean median min max
category                                      
baby products          2.424107    2.0   1   4
baked goods            2.458239    2.0   1   4
baking                 2.454545    2.0   1   4
beverages              2.578073    3.0   1   4
canned foods           2.510441    2.0   1   4
cheese                 2.672355    3.0   1   4
cleaning products      2.482877    2.0   1   4
condiments and sauces  2.486188    2.0   1   4
dairy                  2.533333    3.0   1   4
frozen                 2.509506    3.0   1   4
fruit                  2.443888    2.0   1   4
kitchen                2.500000    2.5   1   4
meat                   2.458115    2.0   1   4
medicine               2.477366    2.0   1   4
packaged foods         2.524655    3.0   1   4
personal care          2.570621    3.0   1   4
pets                   2.453416    2.0   1   4
refrigerated items     2.505882    3.0   1   4
seafood                2.644269    3.0   1   4
snacks                 2.479087    3.0   1   4
spices and herbs       2.432000    3.0   1   4
vegetables             2.510638    3.0   1   4

This line of code outputs a summary statistics for items' volume per category by mean, minimum, and maximum values. Clearly, the mean values are not optimal for our study because the are eveident outliers in the dataset.